### PR TITLE
remove usage of SCFDialect

### LINKED ISSUE #36: Remove SCFDialect if possible
c.f. https://github.com/nebulastream/nebulastream-public/pull/276#discussion_r1751487494

### PR DESCRIPTION
**Pull request by fwc**
Closes #36

### CODE REVIEW

File: nautilus/src/nautilus/compiler/backends/mlir/MLIRPassManager.hpp [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace nautilus::compiler::mlir {
 // and applies configured lowering & optimization passes to it.
 class MLIRPassManager {
 public:
-	enum class LoweringPass : uint8_t { SCF, LLVM };
+	enum class LoweringPass : uint8_t { LLVM };
```
Comment (alepping) [inline nautilus/src/nautilus/compiler/backends/mlir/MLIRPassManager.hpp:12]: Do we really need this enum anymore?

**Review by alepping — COMMENTED** [review]
Can we get rid of the 'LoweringPass' enum?

**Review by alepping — APPROVED** [review]
It would be possible to get rid of the `getMLIRLowerpingPass` function, but since we should do these changes in the nebulastream/nautilus repo, it really is fine the way it is now.

File: nautilus/src/nautilus/compiler/backends/mlir/MLIRPassManager.hpp (hunk 1 above)
Comment (PhilippGrulich) [reply nautilus/src/nautilus/compiler/backends/mlir/MLIRPassManager.hpp:12]: I think we can remove it

**Review by PhilippGrulich — COMMENTED** [review]
This looks good to me. Please remove the full enum. We can also remove the Loop IR node.

**Pull request merged**